src/toil/jobStores/abstractJobStore.py

```diff
@@ -476,102 +488,113 @@ def getJob(jobId):
                 return self.load(jobId)
 
         def haveJob(jobId):
+            assert len(jobId) > 1, "Job ID {} too short; is a string being used as a list?".format(jobId)
```

Comment:
Should this check type?

Reply:
When I caused the kind of problem I'm trying to check against here, `haveJob` was being called with single-character strings, because elsewhere I had used a single job ID string where I meant to have used a list of them. So a type check here in `haveJob` wouldn't help, and if I was smart enough to put a type check where the problem was, I wouldn't have caused it.

Reply:
Python weirdly doesn't distinguish between characters and strings as types.